packages/gafl-webapp-service/src/handlers/agreed-handler.js

```diff
@@ -212,6 +212,10 @@ export default async (request, h) => {
     if (next) {
       return h.redirect(next)
     }
+  } else {
+    await request.ga
+      .ecommerce()
+      .purchase(getTrackingProductDetailsFromTransaction(transaction), '', getAffiliation(process.env.CHANNEL))
```

Comment:
I'm pretty sure that the payment-id is a mandatory field in order for GA to record the transaction as an ecommerce goal.  In the old service we simply used the session id here.

Reply:
doneski